### ISSUE TITLE
[ban-ts-comment] Seemingly valid configuration is being flagged as invalid

### ISSUE THREAD

**Issue by kaiyoma** [typescript-eslint/typescript-eslint#2157]
**Repro**

```JSON
{
  "rules": {
    "@typescript-eslint/ban-ts-comment": [
      "error",
      {
        "ts-ignore": "allow-with-description",
        "minimumDescriptionLength": 10
      }
    ]
  }
}
```

**Expected Result**
The config should be valid.

**Actual Result**

```
Oops! Something went wrong! :(

ESLint: 7.1.0

Error: .eslintrc.json#overrides[0] » eslint-config-arista-ts:
        Configuration for rule "@typescript-eslint/ban-ts-comment" is invalid:
        Value {"ts-ignore":"allow-with-description","minimumDescriptionLength":10,"ts-expect-error":true,"ts-nocheck":true,"ts-check":false} should NOT have additional properties.

    at validateRuleOptions (...\node_modules\eslint\lib\shared\config-validator.js:132:19)
    at ...\node_modules\eslint\lib\shared\config-validator.js:187:9
    at Array.forEach (<anonymous>)
    at validateRules (...\node_modules\eslint\lib\shared\config-validator.js:184:30)
    at validateConfigArray (...\node_modules\eslint\lib\shared\config-validator.js:312:9)
    at CascadingConfigArrayFactory._finalizeConfigArray (...\node_modules\eslint\lib\cli-engine\cascading-config-array-factory.js:464:13)
    at CascadingConfigArrayFactory.getConfigArrayForFile (...\node_modules\eslint\lib\cli-engine\cascading-config-array-factory.js:275:21)
    at FileEnumerator._iterateFilesRecursive (...\node_modules\eslint\lib\cli-engine\file-enumerator.js:431:49)
    at _iterateFilesRecursive.next (<anonymous>)
    at FileEnumerator.iterateFiles (...\node_modules\eslint\lib\cli-engine\file-enumerator.js:287:49)
```

**Versions**

| package                            | version |
| ---------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin` | `3.0.2` |
| `@typescript-eslint/parser`        | `3.0.2` |
| `TypeScript`                       | `3.9.3` |
| `ESLint`                           | `7.1.0` |
| `node`                             | `12.16.1` |
| `npm`                              | `6.13.4` |

**Comment by bradzacher** [comment]
You need to update to 3.1.0 of our tooling.
The docs are always for the latest version of our packages.

**Comment by kaiyoma** [comment]
Ah yes, that was it. Thanks!
